### PR TITLE
torch transforms - RandRicianNoise, StdShiftIntensity, RandStdShiftIntensity

### PR DESCRIPTION
**Pull request by rijobro**
### Description
torch transforms - RandRicianNoise, StdShiftIntensity, RandStdShiftIntensity

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] New tests added to cover the changes.
- [x] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests`.

### CODE REVIEW

**Comment by rijobro** [comment]
@wyli I continue to be hampered by the `NdarrayTensor` problem. Could we come up with a solution that doesn't require `# type: ignore` all over the place? There's not much point in type checking if we then have to use lots of ignores. 

FYI, `NdarrayTensor = Union[np.ndarray, torch.Tensor]` would work for me. Although I'm sure it would break things elsewhere.

**Review by Nic-Ma — APPROVED** [review]
Looks good to me except for some minor comments.
@wyli Do you have any other comments?

Thanks.

**Comment by rijobro** [comment]
@wyli temporarily using `NdarrayTensorUnion = Union[np.ndarray, torch.Tensor]`. Happy to revert in the future once we find a solution.

**Comment by rijobro** [comment]
Thanks @Nic-Ma , updated for both those suggestions.

**Review by wyli — APPROVED** [review]

File: monai/transforms/intensity/array.py [hunk 1]
```diff
@@ -268,22 +284,32 @@ def __init__(
...
+        if isinstance(img, torch.Tensor):
+            ones = torch.ones
+            std = partial(torch.std, unbiased=False)
+        else:
+            ones = np.ones
+            std = np.std
+
+        slices = (img != 0) if self.nonzero else ones(img.shape, dtype=bool)
+        if slices.any():
+            out = deepcopy(img)
```
[the remark this answers is not in the document]
Comment (rijobro) [reply monai/transforms/intensity/array.py]: @Nic-Ma @wyli  are we sure about the in-placeness of the transforms? When I deleted the deepcopy, i needed to modify the unit tests for them to pass - https://github.com/Project-MONAI/MONAI/pull/2815/commits/2431dbaf0a1099c91f6b9e612827df7d525ef82d.

[the remark this answers is not in the document]
Comment (wyli) [reply monai/transforms/intensity/array.py]: you're right, looks like a bug here, this astype will copy if the datatypes doesn't match, if that happens it's not an in-place operation https://github.com/Project-MONAI/MONAI/blob/c39f6fa6414d608b580cbdbdeed48a77c302be3a/monai/transforms/intensity/array.py#L283

**Pull request merged**